### ISSUE TITLE
Possible to make changing logos and titles a config?

### ISSUE THREAD

**Issue by jmagly** [IntegroLabs/hicetnunc#21]
Mirror providers are going to always be a thing, branding may be something that people have varying opinions on, regardless of these its best to make branding name metadata a configuration rather than hard code. 

What would it take to get such a config in place?

**Comment by melMass** [comment]
> What would it take to get such a config in place?

While looking into the rotatin logos ideas with @SlinkousArt yesterday, I realised that we use create-react app which relies on `react-scripts`, AFAIK we cannot further edit the webpack config without `ejecting` (this basicaly extracts all the config files from react-scripts, not something we want I think)

One solution that I've used successfully in the past was using [Craco](https://github.com/gsoft-inc/craco).
This would allow us to inject anything in the webpack config and one could do:

```
import dark_logos from "./logos/dark/*.png"
```

**Comment by melMass** [comment]
Re-reading the question I think I misinterpreted it, you mean a way for forks to easily apply their branding?

**Comment by jmagly** [comment]
Im ok with the configuration being something injected for the build, I dont think its necessarily a requirement for us to make it 100% easy to do at runtime, any mirror will likely be doing a build.

**Comment by jmagly** [comment]
correct, if the concern is a mirror running a deployment appearing to be of one site or another being able to change the branding is a must for anyone running a mirror so its clear the user is on a mirror

**Comment by xat** [comment]
this could be done with the help of environment variables. create-react-app replaces environment variables in the frontend code if they are prefixed with `REACT_APP_`.

see https://create-react-app.dev/docs/adding-custom-environment-variables/

**Comment by xat** [comment]
although if we are talking about multiple things that can be configured, a config file might be the better option.

**Comment by melMass** [comment]
> this could be done with the help of environment variables. create-react-app replaces environment variables in the frontend code if they are prefixed with `REACT_APP_`.

Great so we could have:

```bash
REACT_APP_FORK_NAME = teia
REACT_APP_FORK_DESCRIPTION = The Eternal Institute of Art
REACT_APP_FORK_URL = https://teia.art
REACT_APP_FORK_TWITTER = @teia_community
REACT_APP_FORK_LOGO = teia_logo.png # so we assume it's in the root of public
```

Not sure if it covers everyting

**Comment by jmagly** [comment]
ENV vars during build should be perfectly acceptable.
